src/main/webapp/app/exam/participate/summary/exam-participation-summary.component.html

```diff
@@ -1,44 +1,38 @@
-            {{ 'artemisApp.exam.examSummary.studentSubmissionTo' | artemisTranslate: { examTitle: studentExam.exam!.title!, studentName: studentExam.user!.name! } }}
+            {{ 'artemisApp.exam.examSummary.studentSubmissionTo' | artemisTranslate: { examTitle: studentExam?.exam?.title, studentName: studentExam?.user?.name } }}
         </ng-container>
         <button id="exportToPDFButton" class="btn btn-primary float-end" (click)="printPDF()">
             <fa-icon [icon]="faPrint"></fa-icon>
             <span jhiTranslate="artemisApp.exam.examSummary.exportPDF">Export PDF</span>
         </button>
     </h2>
 </div>
-<jhi-exam-information [exam]="studentExam.exam!" [studentExam]="studentExam"></jhi-exam-information>
-<div class="mb-4 text-center" *ngIf="studentExam && studentExam.exam && isBeforeStudentReviewEnd() && isAfterStudentReviewStart()">
+<jhi-exam-information *ngIf="studentExam?.exam" [exam]="studentExam.exam!" [studentExam]="studentExam"></jhi-exam-information>
+<div class="mb-4 text-center" *ngIf="studentExam?.exam && isBeforeStudentReviewEnd() && isAfterStudentReviewStart()">
     <a class="btn btn-success" style="cursor: default; pointer-events: none">
         {{ 'artemisApp.exam.studentReviewEnabled' | artemisTranslate }}
     </a>
 </div>
 <div class="mb-4 container exam-points-summary-container" *ngIf="studentExam && studentExam.exercises && studentExam.exam && studentExam.exam.course">
-    <jhi-exam-points-summary [exercises]="studentExam.exercises" [exam]="studentExam.exam"></jhi-exam-points-summary>
+    <jhi-exam-points-summary [studentExamWithGrade]="studentExamWithGradeDTO" [exercises]="studentExam.exercises" [exam]="studentExam.exam"></jhi-exam-points-summary>
 </div>
 <div *ngIf="!resultsPublished" class="mb-0">
     <fa-icon [icon]="faInfoCircle" class="info-icon"></fa-icon>
     {{ 'artemisApp.exam.resultInformation' | artemisTranslate }}
 </div>
-<div *ngFor="let exercise of studentExam.exercises; let i = index">
+<div *ngFor="let exercise of studentExam?.exercises; let i = index">
     <div class="question card">
         <div class="question-options card-header question-card-header">
             <h5>
-                &nbsp;<fa-icon *ngIf="exercise && exercise.type" [icon]="getIcon(exercise.type)"></fa-icon>&nbsp;
+                &nbsp;<fa-icon *ngIf="exercise?.type" [icon]="getIcon(exercise.type!)"></fa-icon>&nbsp;
                 <span jhiTranslate="artemisApp.exam.exercise" [translateValues]="{ nr: i + 1 }">Exercise {{ i + 1 }}</span>
             </h5>
-            <div
-                *ngIf="
-                    exercise.studentParticipations?.length &&
-                    exercise.studentParticipations![0].submissions?.length &&
-                    exercise.studentParticipations![0].submissions![0].type === SUBMISSION_TYPE_ILLEGAL
-                "
-            >
+            <div *ngIf="exercise.studentParticipations?.[0]?.submissions?.[0]?.type === SUBMISSION_TYPE_ILLEGAL">
```

Comment:
Same comment as above. Make 100% sure with tests that such simplifications of the code are 100% correct. There was a reason why developers made it differently before.

Reply:
I think the code is equivalent, just uses optional chaining instead of if statements. There are some [special cases ](https://p42.ai/blog/2022-05-04/javascript-refactoring-gotchas-5-ways-converting-to-optional-chaining-can-break-your-code) to keep in mind, but they do not seem to apply here. In my opinion this makes it more readable, but it certainly comes to personal preference.
CC @bassner, our JS expert.

Reply:
I talked with @bassner about the optional chaining related changes and I am forwarding relevant parts of his findings here:

> Your code will work.
The reason for this mass of additional conditions is that the code snippets were just written by someone who (while they were aware of optional chaining this time) did not know that it’s perfectly fine to access an array index that is out of bounds, as, like said above, JS will just yield undefined.

Reply:
Same as above, added unit tests for this simplification at commit https://github.com/ls1intum/Artemis/pull/4932/commits/9896b87f6e9769b29ac20b428dd0bfcf84dc56d2.